packages/template-retail-react-app/app/components/product-tile/index.jsx

```diff
@@ -56,36 +62,87 @@ export const Skeleton = () => {
  * It also supports favourite products, controlled by a heart icon.
  */
 const ProductTile = (props) => {
-    const intl = useIntl()
     const {
-        product,
+        dynamicImageProps,
         enableFavourite = false,
+        imageViewType = 'large',
         isFavourite,
         onFavouriteToggle,
-        dynamicImageProps,
+        product,
+        selectableAttributeId = 'color',
         ...rest
     } = props
+    const {currency, imageGroups, price, productId, hitType} = product
+
+    const intl = useIntl()
+    const {buildUrl} = useMultiSite()
+    const {currency: activeCurrency} = useCurrency()
+    const isFavouriteLoading = useRef(false)
+    const styles = useMultiStyleConfig('ProductTile')
 
-    const {currency, image, price, productId, hitType} = product
+    const isMasterVariant = ['master', 'variant'].includes(hitType)
+    const initialVariationValue = isMasterVariant
+        ? product?.variants?.find((variant) => variant.productId == product.representedProduct.id)
+              ?.variationValues?.[selectableAttributeId]
+        : undefined
+
+    const [selectableAttributeValue, setSelectableAttributeValue] = useState(initialVariationValue)
+
+    const image = useMemo(() => {
+        // NOTE: If the selectable variation attribute doesn't exist in the products variation attributes
+        // array, lets not filter the image groups on it. This ensures we always return an image for non-variant
+        // type products.
+        const hasSelectableAttribute = product?.variationAttributes?.find(
+            ({id}) => id === selectableAttributeId
+        )
+
+        const variationValues = {[selectableAttributeId]: selectableAttributeValue}
+        const filteredImageGroups = filterImageGroups(imageGroups, {
+            viewType: imageViewType,
+            variationValues: hasSelectableAttribute ? variationValues : {}
+        })
+
+        // Return the first image of the first group.
+        return filteredImageGroups?.[0]?.images[0]
+    }, [product, selectableAttributeId, selectableAttributeValue, imageViewType])
+
+    const productUrl = useMemo(() => {
+        return rebuildPathWithParams(productUrlBuilder({id: productId}), {
+            [selectableAttributeId]: selectableAttributeValue
+        })
+    }, [product, selectableAttributeId, selectableAttributeValue])
+
+    // NOTE: variationAttributes are only defined for master/variant type products.
+    const variationAttributes = useMemo(() => {
+        // NOTE: Decorate the product variant attributes to easily access images and hrefs.
+        return product?.variationAttributes?.map((variationAttribute) => ({
```

Comment:
This logic here looks familiar with the useVariantionAttribute.js logic, can we extract or refactor that hook to accommodate both PLP and PDP. If not, can we extract this logic into it owns util function?

Reply:
So I made a utility that we can use in the future and use to refactor what we are already using in the hooks. At this time, that hook is too deeply engrained that I would prefer to do that work outside of this PR as it could possible bloat this PR to the point where the intended work gets hidden in a refactor.